src/sklearndf/pipeline/_learner_pipeline.py

```diff
@@ -187,11 +187,17 @@ def _get_n_features_in(self) -> int:
         else:
             return self.final_estimator.n_features_in_
 
+    def _get_outputs(self) -> Optional[List[str]]:
+        if self.preprocessing is not None:
+            return self.preprocessing._get_outputs()
+        else:
+            return self.final_estimator._get_outputs()
+
     def _get_n_outputs(self) -> int:
         if self.preprocessing is not None:
             return self.preprocessing.n_outputs_
         else:
-            return self.final_estimator.n_outputs_
+            return self.final_estimator._get_n_outputs()
```

Comment:
@j-ittner Shouldn't we have same method call internally for `_get_n_outputs` as in `_get_outputs`? For preprocessing it's n_outputs_ but for final_estimator it's _get_n_outputs. I would leave n_outputs as it's the same with `is_fitted` check. What do you think?

Reply:
Well spotted. It should be `_get_n_outputs()` in both branches, because once we're in this method, we have already checked that all is fitted.